cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -478,6 +478,38 @@ std::shared_ptr<PointCloud> PointCloud::RandomDownSample(
     return SelectByIndex(indices);
 }
 
+std::shared_ptr<PointCloud> PointCloud::GetFarthestPointSample(
+        size_t num_samples) const {
+    if (num_samples <= 0) {
+        utility::LogError(
+                "[GetFarthestPointSample] Illegal number of points, the number "
+                "should larger than 0");
+    }
+    if (num_samples >= points_.size()) {
+        return std::make_shared<PointCloud>(*this);
+    } else {
+        std::vector<size_t> indices;
+        indices.resize(num_samples);
+
+        const size_t num = points_.size();
+        std::vector<double> distance(num, 1e10);
+        size_t farthest_index = 0;
+        for (size_t i = 0; i < num_samples; i++) {
+            indices[i] = farthest_index;
+            auto &selected = points_[farthest_index];
```

Comment:
Let's be more explicit: `auto &selected` -> `Eigen::Vector3d & selected_point`. 
Also `const`?